src/point_one/fusion_engine/parsers/fusion_engine_framer.cc

```diff
@@ -189,14 +200,23 @@ p1_ssize_t FusionEngineFramer::OnByte(bool quiet) {
       // Compute the full message size. If the message is too large to fit in
       // the buffer, we cannot parse it. Otherwise, start collecting the
       // message payload.
+      //
+      // Note that while we compute the full message size here, we intentionally
+      // do the "too big" check below with the payload size. On a 32b machine,
+      // if the 32b payload size value is max (0xFFFFFFFF) or close to it,
+      // current_message_size_ will overflow and it will look like the message
+      // is _not_ too big. Normally, this won't happen for legit packets that
+      // are just too big for the user's buffer, but it could happen on a bogus
+      // header if we find the preamble randomly in an incoming byte stream.
       auto* header = reinterpret_cast<MessageHeader*>(buffer_);
       current_message_size_ =
           sizeof(MessageHeader) + header->payload_size_bytes;
       VLOG(3) << "Header complete. Waiting for payload. [message="
               << header->message_type << " (" << (unsigned)header->message_type
               << "), seq=" << header->sequence_number
               << ", payload_size=" << header->payload_size_bytes << " B]";
-      if (current_message_size_ <= capacity_bytes_) {
+      if (header->payload_size_bytes <=
```

Comment:
It seems safer to check both conditions (old and new). In general I like.bounds checks to be directly on the variables indexing those bounds, not by a derivative quantity in case other logic is changed later.

Reply:
Hmm I suppose. Seems like a wasted instruction though since this is the literal definition of the message size. Normally I would have just moved the size calculation to inside the if statement, but we print it in the warning message so I left it here.

Reply:
ok that's fair, the variable names confused me earlier, I see its basically the same check.